source/exe/signal_action.cc

```diff
@@ -57,13 +59,24 @@ void SignalAction::installSigHandlers() {
 }
 
 void SignalAction::removeSigHandlers() {
+#if defined(__APPLE__)
+  // ss_flags contains SS_DISABLE, but Darwin still checks the size, contrary to the man page
+  if (previous_altstack_.ss_size < MINSIGSTKSZ) {
+    previous_altstack_.ss_size = MINSIGSTKSZ;
+  }
+#endif
   RELEASE_ASSERT(sigaltstack(&previous_altstack_, nullptr) == 0);
+
   int hidx = 0;
   for (const auto& sig : FATAL_SIGS) {
     RELEASE_ASSERT(sigaction(sig, &previous_handlers_[hidx++], nullptr) == 0);
   }
 }
 
+#ifdef __APPLE__
+#define MAP_STACK (0)
```

Comment:
Does it make sense to add #ifndef MAP_STACK around this in case Darwin adds real map_stack in the future?